melee/SSBMEnv.py

```diff
@@ -359,23 +395,31 @@ def step(self, joint_action): # step should advance our state (in the form of th
             raise RuntimeError("Controller port inconsistency!")
 
         prev_gamestate = self.gamestate
-        self.state_data.append({'state': self._get_state()})
+
         # perform actions
         for agent_idx, agent in enumerate(self.agents):
             action = joint_action[agent]
-            self.state_data[-1][agent] = action
-            self._perform_action(agent_idx, action)
+            self._perform_action(agent_idx, action)      
 
         # step env
         self.gamestate = self.console.step()
-        # collect reward
+        
+        # Collect all transition data
         reward = self.get_reward(prev_gamestate, self.gamestate)
         state = self._get_state()
-        self.state_data[-1]['next_state'] = state
-        # determine if game is over and write extra info
         done = self._get_done()
         info = self._get_info()
 
+        # Log (s_t, a_t, s'_t) data if necessary
+        if self.dump_states:
+            # Log a_t
+            for agent in self.agents:
+                self.state_data[-1][agent] = joint_action[agent]
+            # Log s'_t
+            self.state_data[-1]['next_state'] = state
+            # Log s_{t+1}
+            self.state_data.append({ "state" : state })
```

Comment:
Need to put a check here that if done, don't append {s}. That way the BC agent only gets complete {s, a, s'} pairs

Reply:
good catch! Went ahead and fixed